### DISCUSSION TITLE
Hot air distribution system at home - simple thermostat rule issue.

### DISCUSSION THREAD

**Discussion by tykasan** [arendst/Tasmota#14030] [Q&A] [answered]
Hello everyone, 
I'd like to ask you for a help on determining, where I made a mistake on rules setup for my simple thermostat

Idea is to use ESP-01 with relay and DS18B20 sensor.
If temperature reach 30C then relay (FAN) would be turned on and hot air from fireplace would be distributed around the house.
Usually temperature reaches around 60C when heating... when heating cycle is over, temperature drops and I need to turn off the FAN , BUT i need to do it when temperature is around 40C

I developed below, but its not working as expected
Fan is turning on when expected, but it doesnt turning off on below 40C

` Rule1
 ON system#boot DO var1 0 ENDON
 ON DS18B20#Temperature>30 DO if ((var1==0) AND (POWER1==0)) POWER1 1; var1 1 endif ENDON
 ON DS18B20#Temperature<40 DO if ((var1==1) AND (POWER1==1)) POWER1 0; var1 2  endif ENDON
 ON DS18B20#Temperature<30 DO Backlog var1 0; POWER1 0 ENDON   `
 
Would you please help me to understand what Im doing wrong?
I compiled custom build tasmota to get features like IF/Else - this part is working well :)

Please advise.

**Comment by tykasan** [comment, marked as answer]
anyone?

What I did additionally, I tried to test IF statement with:

	 `Rule1 ON time#minute DO if (DS18B20#Temperature>40) POWER1 1 endif endon `
	 
This should turn on relay when temperature is higher then 40C

with result:

`23:17:00.430 RUL: TIME#MINUTE performs "if (DS18B20#Temperature>40) POWER1 1 endif"
23:17:00.437 RSL: RESULT = {"If":"Done"}`	 

So it looks it process if statement, but relay was not turned on...

I suspect there might be something with IF statement issue.

What are your thoughts?

**Reply by barbudor** [reply]
You totally mis understand how rules are working
Read again the documentation
Rules are not a programming language
https://tasmota.github.io/docs/Rules/

- You cannot "test" POWER : `if ((var1==0) AND (POWER1==0))` what is after AND is wrong
- You cannot "test" a trigger : `if (DS18B20#Temperature>40)` this is wrong
- A trigger can only be located between `ON` and `DO`
- You can get the value attached to the trigger with `%value%`

You may also read the section about the Thermostat feature which will avoid you to re-invent the wheel
https://tasmota.github.io/docs/Thermostat/

**Reply by sfromis** [reply]
There is also a simpler example of how you can do a very basic thermostat function in rules:
https://tasmota.github.io/docs/Rules/#simple-thermostat-example

**Reply by tykasan** [reply]
Thanks Barbudor, Sformis
I read manual once again.... now I got it... :) 
Sometimes I need a heavy kick in the back part of my body to get things done.
I made it working wih few additional variables and rules...

Thanks guys!
